protocols/dcutr/src/handler/relayed.rs

```diff
@@ -247,15 +246,24 @@ impl Handler {
                     },
                 ));
             }
-            _ => {
-                // Anything else is considered a fatal error or misbehaviour of
-                // the remote peer and results in closing the connection.
-                self.pending_error = Some(error.map_upgrade_err(|e| {
-                    e.map_err(|e| match e {
-                        EitherError::A(e) => EitherError::A(e),
-                        EitherError::B(v) => void::unreachable(v),
-                    })
-                }));
+            ConnectionHandlerUpgrErr::Upgrade(UpgradeError::Select(
+                NegotiationError::ProtocolError(p),
+            )) => {
+                self.queued_events
+                    .push_back(ConnectionHandlerEvent::Close(CloseReason::new(
+                        "dcutr-relay",
+                        p,
+                    )));
```

Comment:
How about `trait CloseReasonExt` to make this `ConnectionHandlerEvent::Close(p.closeReason("dcutr-relay"))`?

Reply:
We could do:

```rust
p.into_close_reason("dcutr-relay")
```

perhaps?

Reply:
Not to start bikesheding but what about impl `From<(&str, Box<dyn Error + Send + 'static>)> for CloseReason`  and have `ConnectionHandlerEvent::Close(("dcutr-relay", p).into())`?

Reply:
Personally, I am not a fan of `From<(.., ..)>` implementations. I generally tend to avoid to implement `From` unless I can provide a function somewhere that accepts `impl Into` and thus allows us to avoid `.into()` calls somewhere. To me, calling `.into` yourself is a bit of a code smell.